File: SCEUCN-APP/app/src/main/java/cl/ucn/disc/pdis/sceucn/model/Vehiculo.java
/*
 * Created by Christian Farias on 22-04-19 2:02
 * Copyright (c) 2019 . All rights reserved.
 * Last modified 22-04-19 2:02
 */

package cl.ucn.disc.pdis.sceucn.model;

import android.arch.persistence.room.Embedded;
import android.arch.persistence.room.Entity;
import android.arch.persistence.room.PrimaryKey;

import java.util.List;

import lombok.Data;
import lombok.NonNull;

@Data
@Entity(tableName="vehiculo")
public class Vehiculo {

    /**
     * La persona a la cual le pertenece este vehiculo.
     */
    @Embedded(prefix = "persona_")
    @NonNull
    private Persona persona;

    /**
     * La placa del vehiculo. Ej: FBXS22.
     */
    @PrimaryKey
    @NonNull
    private String placa;

    /**
     * El anio de...?
     */
    private String anio;

    /**
     * La marca del vehiculo. Ej: Suzuki.
     */
    private String marca;

    /**
     * El tipo de vehiculo. Ej: Auto.
     */
    private Tipo tipo;

    /**
     * Los logos de este vehiculo.
     */
    @Embedded(prefix = "logo_")
    private List<Logo> logos;
}

File: SCEUCN-SERVER/Model/Vehiculo.cs

using System.Collections.Generic;

namespace CL.UCN.DISC.PDIS.SCE.Server.Model
{
    public enum Tipo
    {
        Auto,
        Camioneta,
        Moto
    }

    public class Vehiculo
    {
        public string Id { get; set; }

        public string Anio { get; set; }

        public string Placa { get; set; }

        public Persona Persona { get; set; }

        public string Marca { get; set; }

        public Tipo Tipo { get; set; }

        public List<Logo> Logos { get; set; }
    }
}